CMakeLists.txt

```diff
@@ -235,12 +239,23 @@ endif()
 
 ########################################################################
 # base64
-add_subdirectory(bin)
+if (BASE64_BUILD_CLI)
+    add_executable(base64-bin 
+        bin/base64.c
+    )
+    target_link_libraries(base64-bin PRIVATE base64)
+    set_target_properties(base64-bin PROPERTIES 
+        OUTPUT_NAME base64
+    )   
```

Comment:
Trailing whitespace, also on lines 243 and 247.

Reply:
Have you heard of https://editorconfig.org/ ? Most editors support it and it has an [option] instructing editors to squelch those pesky trailing spaces automatically.

[option]: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#trim_trailing_whitespace

Reply:
@BurningEnlightenment Never heard of it until now. I'm an oldschool guy who runs `vim` in a terminal and exclusively uses `git` through the command line interface, maybe that's why :)

I'm not a huge fan of things like modelines that mess with people's editor settings, and it looks like this might fall in that category. But as maintainer of this project, I do insist on adherence to a certain formatting, such as tabs for indent, no trailing whitespace, and so on. If an `.editorconfig` can politely warn the user about things that run counter to the intended formatting, then I do see the value of adding that file. Until then, I'm also fine with doing a manual style check before merging.

I've also considered setting up a style checker in CI (something like `uncrustify` or `clang-format`), but I think it would be a bit annoying, because people would need to run the tool locally to make sure that their code passes CI.

@htot No worries, that's what code review is for :)

Reply:
Oh, I'm also a CLI person, but I seldomly use vim because that is/was not so well supported on Windows.

I'm not familiar with vim `modelines`, but an `.editorconfig` can be seen as a very basic but universal `.clang-format`. It instructs editors to (temporarely) apply the configured formatting settings (only) within the containing directory. It only supports the options `indent_style`, `tab_width`, `end_of_line` (style), `charset`, `trim_trailing_whitespace`, `insert_final_newline`, but as opposed to `.clang-format` it is more widely applicable and natively supported by editors. People who get disturbed by this usually have disabled/not installed editorconfig support globally.

Adding a `.clang-format` is a good idea, too. Quite a few editors have plugins/integrations for enforcing it on save (even [vim]), so chances are that people wouldn't need to manually run `clang-format`. However, `clang-format` can't format CMake files.

And just for the record there is an [editorconfig plugin] for vim, too 😉 

[editorconfig plugin]: https://github.com/editorconfig/editorconfig-vim#readme
[vim]: https://github.com/rhysd/vim-clang-format

Reply:
@BurningEnlightenment Thanks for explaining, I created #81 to add an `.editorconfig` file.